api/base/serializers.py

```diff
@@ -79,6 +79,14 @@ def to_internal_value(self, data):
         return super(TypeField, self).to_internal_value(data)
 
 
+class JSONAPIListField(ser.ListField):
+    def to_internal_value(self, data):
+        if not isinstance(data, list):
+            self.fail('not_a_list', input_type=type(data).__name__)
+
+        return super(JSONAPIListField, self).to_internal_value(data)
```

Comment:
This appears to be a bug in Django Rest Framework's ListField.

```python
    def to_internal_value(self, data):
        #...
        if isinstance(data, type('')) or not hasattr(data, '__iter__'):
                self.fail('not_a_list', input_type=type(data).__name__)
```

It is not properly checking if `data` is dict/mapping.

@pattisdr Would you like to report this issue and/or send a PR to DRF?